pkg/reconcile/v2/runner.go

```diff
@@ -146,7 +146,7 @@ func (r *runner[T]) RunReconciler(
 	}
 
 	// send us watch events
-	if err := ctl.Watch(&source.Kind{Type: obj}, &handler.EnqueueRequestForObject{}, predicates...); err != nil {
+	if err := ctl.Watch(source.Kind(r.mgr.GetCache(), obj), &handler.EnqueueRequestForObject{}, predicates...); err != nil {
```

Comment:
nice it mimicks upstream's change transparently! https://github.com/kubernetes-sigs/controller-runtime/commit/d6a053f8f0979d6f1528fe4d52b0d0f618352d77#diff-fa148e22912d4a8390270dd209fd4c81fad957622efe2d0dad8a5455d0589201R62

That being said are we sure that mgr is not nil?

Reply:
I felt safe making this change, because earlier in this function there is:
```
gvk, err := apiutil.GVKForObject(obj, r.mgr.GetScheme())
```

Reply:
yeah exactly